src/components/axes/abstractAxis.ts

```diff
@@ -488,14 +488,40 @@ export module Axis {
                                     });
       var lastLabelClientRect: ClientRect;
 
-      visibleTickLabels.each(function (d: any) {
+      // Get the widths of all the labels
+      var visibleTickLabelRects: ClientRect[] = [];
+      visibleTickLabels.each(function() {
         var clientRect = this.getBoundingClientRect();
+        visibleTickLabelRects.push(clientRect);
+      });
+
+      // Figure out the appropriate interval for overlap
+      var hasSkipped = false;
+      var interval = 1;
+      // Only do interval modifications for bottom and top orientations
+      if (this._orientation === "bottom" || this._orientation === "top") {
+        for (var i = 1; i < visibleTickLabelRects.length; i++) {
+          // Checks to see if there is any overlap between the first element and the current element
+          if (!_Util.DOM.boxesOverlap(visibleTickLabelRects[i], visibleTickLabelRects[0])) {
```

Comment:
Isn't there still a potential problem here, where the first label's rectangle is very small, but later labels overlap each other?